### PR TITLE
Support Metadata in GridFSAdapter

### PR DESCRIPTION
**Pull request by dplewis**
* Useful for testing in the JS SDK
* Adds new endpoint to be used with `Parse.File.getData`
* Allows file adapters to return tags as well as future data.

### CODE REVIEW

**Comment by dplewis** [comment]
I'm trying to improve coverage but I can't log into codecov. I've been trying for 2 days.

@acinader @davimacedo Are you able to log in?

Screen Shot 2020-05-01 at 10 20 35 PM

File: spec/GridFSBucketStorageAdapter.spec.js [hunk 1]
```diff
@@ -33,6 +35,73 @@ describe_only_db('mongo')('GridFSBucket and GridStore interop', () => {
...
+    await gfsAdapter.createFile('myFileName', originalString, null, {
+      metadata,
+    });
+    const gfsResult = await gfsAdapter.getFileData('myFileName');
+    expect(gfsResult.toString('utf8')).toBe(originalString);
+    let gfsMetadata = await gfsAdapter.getMetadata('myFileName');
+    expect(gfsMetadata.metadata).toEqual(metadata);
+
+    // Empty json for file not found
+    gfsMetadata = await gfsAdapter.getMetadata('myUnknownFile');
```
Comment (acinader) [inline spec/GridFSBucketStorageAdapter.spec.js:51]: should really be a separate test, right?

**Comment by acinader** [comment]
@dplewis do you have an access issue with codecov I can help with?

**Comment by dplewis** [comment]
@acinader Codecov works now. I'm increasing test coverage here. The coverage diff has nothing to do with this PR. Most likely Prettier

**Pull request merged**